File: src/main/java/com/github/svegon/capi/event/ListenerSortedSet.java
package com.github.svegon.capi.event;

import java.util.SortedSet;
import java.util.function.Function;

public class ListenerSortedSet<L> extends ListenerCollection<L, SortedSet<L>> {
    ListenerSortedSet(SortedSet<L> listeners, Function<SortedSet<L>, L> invokerFactory) {
        super(listeners, invokerFactory);
    }
}

File: src/main/kotlin/io/github/svegon/mclientapi/event/ListenerSortedSet.kt
package io.github.svegon.mclientapi.event

import java.util.*

class ListenerSortedSet<L> internal constructor(listeners: SortedSet<L>,
                                                invokerFactory: (SortedSet<L>) -> L) :
    ListenerCollection<L, SortedSet<L>>(listeners, invokerFactory)
